kotsadm/pkg/downstream/downstream.go

```diff
@@ -431,6 +480,29 @@ func versionFromRow(appID string, row scannable) (*types.DownstreamVersion, erro
 	return v, nil
 }
 
+func preflightResultFromRow(row scannable) (*types.PreflightResult, error) {
+	r := &types.PreflightResult{}
+
+	var preflightResult sql.NullString
+	var preflightResultCreatedAt sql.NullTime
+
+	if err := row.Scan(
+		&preflightResult,
+		&preflightResultCreatedAt,
+		&r.AppSlug,
+		&r.ClusterSlug,
+	); err != nil {
+		return nil, errors.Wrap(err, "failed to scan")
```

Comment:
Is it possible that this returns NoRows, which would cause an error to be rendered in the UI?

Reply:
But i think in this case we want an error. Probably better to have a 404 and a message...

<img width="1128" alt="Screen Shot 2020-08-18 at 4 29 45 PM" src="https://user-images.githubusercontent.com/371319/90577374-91195080-e175-11ea-82e8-3b995ca9f6ec.png">